### DISCUSSION TITLE
Hide Bot Token in GitHub

### DISCUSSION THREAD

**Discussion by BradTFT** [Rapptz/discord.py#7806] [Q&A] [answered]
Hi community,

Im using github as version control (some people asked my why im uploading my bots code to github) and i need a way to hide my token in my repo. is there any way to do this. ive seen people do things with other files but idk how to make sure that the last line of my bot.py file so it runs. anyway, i just need a way to censor my token in my repo

**Comment by Vexs** [comment]
You can't* preclude a specific line of a specific file from being committed. Store your token in another file or in environment variables or similar.

(*you can, but it's stupid)

**Reply by BradTFT** [reply]
im using python. how can i link the files and still have the functionality of my main file

**Comment by Chrovo** [comment]
Assuming you have a file to store your token in, create another file called .gitignore. Put the name of the file(the file that your token is being stored in) in the .gitignore file. This should block your secret file from being viewed publicly.

**Reply by BradTFT** [reply]
how do i connect the token to the main file? or do i have to remove the token line every time i want to commit

**Reply by Chrovo** [reply]
Lets say you are storing your token in a JSON file. You can import the json module, use context managers and use the json.load method and then access the token from there.

**Reply by BradTFT** [reply]
cool. any chance you can point me in a direction to do this. like a tutorial.

**Reply by JDJGInc** [reply]
you are better off asking this in the discord.gg/dpy guild.

**Reply by BradTFT** [reply]
thanks for the info

**Reply by BradTFT** [reply]
@JDJGInc link is expired

**Reply by scrazzz** [reply]
> @JDJGInc link is expired

discord.gg/dpy

**Reply by BradTFT** [reply]
@scrazzz appreciate it thanks

**Reply by BradTFT** [reply]
just realized i am already in the discord. they pointed me here. said it was a better fit to ask on github

**Comment by scrazzz** [comment]
Why don't you make your repository private? That way you can keep your token safe without worrying about someone else finding it.

**Reply by BradTFT** [reply]
this would be more of an education thing for me to learn how to hide the token. but yes that is an option

**Comment by GoogolGenius** [comment, marked as answer]
Use can use environment variables with Python dotenv to keep the token secret. And in your .gitignore file you would omit it. [Read on the PyPi website.](https://pypi.org/project/python-dotenv/#getting-started)

**Comment by trevorflahardy** [comment]
This was already answered in the help channel at 3:31pm EST today. You can review that [here](https://discord.com/channels/336642139381301249/564950631455129636/958448391082356757) if interested.

 The help channel is always there for you, feel free to @ me in it anytime - Chai#9762. It's always going to be faster to ask in the help channel than ask in GitHub discussions :)
